tests/integration/components/au-radio-group-test.js

```diff
@@ -0,0 +1,87 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'dummy/tests/helpers';
+import { click, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | au-radio-group', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it can be given a name', async function (assert) {
+    await render(hbs`
+      <AuRadioGroup @name="foo" as |Group|>
+        <Group.Radio data-test-foo>Foo</Group.Radio>
+      </AuRadioGroup>
+    `);
+    assert.dom('[data-test-foo]').hasAttribute('name', 'foo');
+  });
+
+  test('if it is given no name, it will automatically generate a unique id', async function (assert) {
+    await render(hbs`
+      <AuRadioGroup as |Group|>
+        <Group.Radio data-test-foo>Foo</Group.Radio>
+      </AuRadioGroup>
+    `);
+    assert.dom('[data-test-foo]').hasAttribute('name', /^ember\d+/); // E.g. 'ember308'
```

Comment:
I wouldn't test it like that since the output an implementation detail. I would just check if the attribute exists since that means _some_ value is set.

A better way could be to check multiple radios and test if only 1 of them stays checked?

Reply:
Sounds more logical to me 👍 I've adjusted the existing test & extended the one after that to cover your last question.